blocks/components/editable/index.js

```diff
@@ -60,55 +60,86 @@ function createElement( type, props, ...children ) {
 }
 
 export default class Editable extends wp.element.Component {
-	constructor() {
+	constructor( props ) {
 		super( ...arguments );
 
 		this.onInit = this.onInit.bind( this );
 		this.onSetup = this.onSetup.bind( this );
-		this.onChange = this.onChange.bind( this );
+		this.onSelectionChange = this.onSelectionChange.bind( this );
 		this.onNewBlock = this.onNewBlock.bind( this );
 		this.onFocus = this.onFocus.bind( this );
 		this.onNodeChange = this.onNodeChange.bind( this );
 		this.onKeyDown = this.onKeyDown.bind( this );
 		this.changeFormats = this.changeFormats.bind( this );
+		this.onChange = this.onChange.bind( this );
+
 		this.state = {
 			formats: {},
-			alignment: null,
-			bookmark: null
+			alignment: null
 		};
+
+		this.change = false;
+		this.content = props.value;
+		this.selection = props.selection;
 	}
 
 	onSetup( editor ) {
 		this.editor = editor;
 		editor.on( 'init', this.onInit );
-		editor.on( 'focusout', this.onChange );
 		editor.on( 'NewBlock', this.onNewBlock );
-		editor.on( 'focusin', this.onFocus );
 		editor.on( 'nodechange', this.onNodeChange );
 		editor.on( 'keydown', this.onKeyDown );
+		editor.on( 'focus', this.onFocus );
+		editor.on( 'change', this.onChange );
+		editor.on( 'selectionChange', this.onSelectionChange );
 	}
 
 	onInit() {
-		this.updateFocus();
+		this.maybeFocus();
 	}
 
 	onFocus() {
 		if ( ! this.props.onFocus ) {
 			return;
 		}
 
-		// TODO: We need a way to save the focus position ( bookmark maybe )
-		this.props.onFocus();
+		this.props.onFocus( this.selection );
 	}
 
 	onChange() {
-		if ( ! this.editor.isDirty() ) {
+		this.change = true;
+	}
+
+	isActive() {
+		return document.activeElement === this.editor.getBody();
+	}
+
+	shouldChange( prevContent ) {
+		return this.isEmpty( prevContent ) !== this.isEmpty( this.content );
+	}
+
+	onSelectionChange() {
+		// We must check this because selectionChange is a global event.
+		if ( ! this.isActive() ) {
 			return;
 		}
 
-		this.savedContent = this.getContent();
-		this.editor.save();
-		this.props.onChange( this.savedContent );
+		// Make sure text nodes are normalized before passing the content and selection.
+		// Normalized means that text nodes are merged were possible.
+		if ( this.change ) {
+			this.editor.getBody().normalize();
+		}
+
+		const prevContent = this.content;
+
+		this.content = this.getContent();
+		this.selection = this.getSelection();
+
+		if ( this.change || this.shouldChange( prevContent ) ) {
```

Comment:
`this.change` will change when the TinyMCE change even is called, or when we manually update this flag. `shouldChange` is based on the previous and current content.